### PR TITLE
DAOS-7920 build: Include pydaos and storage estimator in installed site packages dir

### PR DESCRIPTION
**Pull request by mjean308**
Signed-off-by: Maureen Jean <maureen.jean@intel.com>

### CODE REVIEW

File: src/client/pydaos/SConscript [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
 
     if GetOption('help'):
         return
 
     new_env = env.Clone()
     extra_flags = ""
-
-    if version == "3":
+    version = "{}.{}".format(
+        sys.version_info.major, sys.version_info.minor)
```
Comment (phender) [inline src/client/pydaos/SConscript:14]: Optional: If you wanted to shorten this, `sys.version_info[0] == sys.version_info.major` and `sys.version_info[1] == sys.version_info.minor`, so you could use a python list slice:
```suggestion
    version = "{}.{}".format(*sys.version_info[0:2])
```

File: utils/rpms/daos.spec [hunk 2]
```diff
@@ -375,11 +375,11 @@ getent passwd daos_agent >/dev/null || useradd -s /sbin/nologin -r -g daos_agent
 %{_libdir}/libdaos_common_pmem.so
 %{conf_dir}/vos_size_input.yaml
 %{_bindir}/daos_storage_estimator.py
-%{_libdir}/python3/site-packages/storage_estimator/*.py
-%dir %{_libdir}/python3/site-packages/storage_estimator
-%if (0%{?rhel} >= 7) && (0%{?rhel} < 8)
-%dir %{_libdir}/python3/site-packages/storage_estimator/__pycache__
-%{_libdir}/python3/site-packages/storage_estimator/__pycache__/*.pyc
+%{_libdir}/python3.6/site-packages/storage_estimator/*.py
```
Comment (phender) [inline utils/rpms/daos.spec:378]: Should we be using `python3_sitelib` here and everywhere else we have `python3.6` hardcoded: 
```suggestion
%{python3_sitelib}/storage_estimator/*.py
```

**Review by phender — CHANGES_REQUESTED** [review]

File: utils/rpms/daos.spec (hunk 2 above)
Comment (mjean308) [reply utils/rpms/daos.spec:378]: Ihad tried using %{python3_sitearch} and it failed I'll try sitelib

**Review by phender — DISMISSED** [review]

**Review by bfaccini — COMMENTED** [review]
Will this handle the case where a new python version >= 3.8 is later installed ? Will pydaos and storage estimator modules still be available ?

**Comment by ashleypittman** [comment]
I'm surprised pydaos isn't in the client rpm already, but I don't think we want to package pydaos.raw do we?

**Comment by mjean308** [comment]
@bfaccini 
A daos build will include pydaos and storage estimator in the correct site-packages dir.  However; the daos rpms will create the site-packages directory based on the version of python that was present on our build server %{python3_sitearch} which maynot be the same as what a customer and would cause the same issue.

**Comment by mjean308** [comment]
@ashleypittman  I can remove it if you think it should be.    I was only getting rid of the "python3/site-packages" and using the build version instead.   but I can cleanup these files if they should not be packaged

**Comment by ashleypittman** [comment]
> @ashleypittman I can remove it if you think it should be. I was only getting rid of the "python3/site-packages" and using the build version instead.

Ah I see now, yes.  I thought you were adding something that wasn't previously included.

**Review by phender — DISMISSED** [review]

**Review by bfaccini — DISMISSED** [review]
sounds good for installs where python3 version will not upgrade, but we need to think about what could be done if this becomes the case...

**Review by ashleypittman — APPROVED** [review]

**Review by shimizukko — APPROVED** [review]

**Review by ashleypittman — APPROVED** [review]

**Review by shimizukko — APPROVED** [review]

**Pull request merged**